src/main/java/auto_scheduler/CoursePrioritizationServiceImpl.java

```diff
@@ -10,16 +13,193 @@
 import java.lang.NullPointerException;
 
 public class CoursePrioritizationServiceImpl implements CoursePrioritizationService {
-
     public Collection<CurriculumCourse> build(
         Collection<CurriculumCourse> candidateCourses, 
         Collection<String> passedCourses) {
```

Comment:
we probably don't need passed classes. but not a 100% sure yet. so i have not removed it.